content/livesync/models/models.textile

```diff
@@ -0,0 +1,278 @@
+
+h3(#history). Replay from history
+
+After the Models SDK initializes the model state via the sync function, it is important to apply the exact sequence of change events from your backend from the correct point in the stream. This position is specified by the @sequenceID@ returned from your backend endpoint.
+
+Internally, after calling the sync function, the SDK attaches to the "channel":/channels and "subscribes":#subscribe to live messages. Live messages are buffered and not yet processed. It then paginates backwards through the message history on the channel from the point of attachment using "@untilAttach@":/storage-history/history#until-attach. The number of items in each page can be configured via the @syncOptions.historyPageSize@ defined in your "@ClientOptions@":/livesync/models#options.
+
+When a message with the target @sequenceID@ is reached, all of the messages from that point forward are sent to the model for processing to maintain continuity with incoming live messages.
+
+If the target @sequenceID@ cannot be reached, there is insufficient message history to resume from the correct point and bring the model up to date. The SDK will try to sync again with a fresher version of the model state by calling your sync function once again.
+
+The amount of history available to query on the channel is determined by your Ably "message storage":/storage-history/storage configuration on the channel. This configuration must match the @syncOptions.messageRetentionPeriod@ defined in your "@ClientOptions@":/livesync/models#options. The SDK uses this configuration option as a hint as to whether to resynchronize via the sync function and skip paginating through history when messages are expected to have expired.
+
+You may wish to cache the model state from the backend endpoint called by the sync function. There is a strict requirement that the model state returned by this endpoint is no older than the message retention period configured on the channel so that the model state can be brought up to date.
+
+If there are no historical messages at all on the channel, then the model state is assumed to be brand new, having not yet undergone any changes.
+
+h2(#merge). Merge functions
+
+A merge function tells your model how to calculate the next version of the model state when a change event is received from the backend and is required when instantiating a model. It is a function that accepts the previous model state and the change event as input and returns the next version of the model state.
+
+
+The following is an example of a merge function. It will be invoked for all events received on the channel name specified in the model:
+
+```[javascript]
+async function merge(state: Post, event: OptimisticEvent | ConfirmedEvent) {
+  if (event.name === 'addComment') {
+     return {
+    ...state,
+    comments: state.comments.concat([event.data]),
+  };
+   }
+ // handle other event types
+}
+```
+
```

Comment:
Same as above, I think we need to show how we register the merge function on the model on instantiation